src/lib/distances.jl

```diff
@@ -47,10 +47,10 @@ end
   end
 
 @adjoint function (::Euclidean)(x::AbstractVector, y::AbstractVector)
-  D = x.-y
+  D = x .- y
   δ = sqrt(sum(abs2, D))
   function euclidean(Δ::Real)
-    x̄ = (Δ / δ) .* D
+    x̄ = (Δ / max(δ, eps(typeof(δ)))) .* D
```

Comment:
So the only time ` δ`  is less than `eps(typeof(δ))` is  is `δ==0`, since this is definition of `eps`.
and that can only occur if `D==[0, 0, 0,...]`.
and in that case we get that `x̄ ` iz also zero, of the same size as `D`.

So we can do:
```suggestion
    x̄ = ifelse(iszero(δ), D, (Δ / δ) .* D)
```